optimize/unconstrained_test.go

```diff
@@ -1180,7 +1180,9 @@ func testLocal(t *testing.T, tests []unconstrainedTest, method Method) {
 		}
 		settings.GradientThreshold = test.gradTol
 
-		result, err := Local(test.p, test.x, settings, method)
```

Comment:
It's a pity that by moving the initial X to settings it becomes optional. I think it is very natural to have it mandatory for local optimization. And even the global methods are not truly global and their result depends on where they are started.

Reply:
I don't think I agree with this, but I'm happy to debate. It seems natural to me to have the starting location be the zero value of the type, and I imagine that's a reasonable default for a lot of problems (I certainly often just have `initX := make([]float64, dim)`. I also don't think it's true that global methods have an initial location. `CmaEsChol` does. `ListOpt` does not, it has a starting population. This would also be true for most genetic algorithms. `GuessAndCheck` has a starting distribution, as do cross-entropy methods.

Reply:
> I imagine that's a reasonable default for a lot of problems

This is the point that I'm not so sure about in practice. Having a mandatory `initX` slice would send a clear message that the starting location determines the local minimum in case there are more of them.

> I also don't think it's true that global methods have an initial location

Here I meant a starting location in a more general sense, that's why I wrote "where they are started". All your examples have a starting something that determines where the minima will be searched for. `ListOpt` is a bit of an outlier, it satisfies `GlobalMethod` but I would not call it a _method_.

Anyway, my comment was just a comment, I'm fine with the change.

Reply:
I agree with your argument of "starting location in a more general sense", but there's no way to convey that in the API, when sometimes it's a vector, and sometimes it's a matrix.

That said, the previous thought was to have `optimize.Minimize(p Problem, dim int, ...)` . Instead of having `dim int`, we could have `initX []float64`. It conveys the needed dimension information, allows us to remove the field from settings, and global methods that don't need it can ignore it.

Reply:
> we could have initX []float64

I'll leave this up to you. I can't make up my mind right now and I don't want to block on this.

Reply:
Well, fortunately that'll be part of the next PR anyway :-)